It seems like Cirrus CI has some restriction on the network. Is it expected? I see that you added ``-u-network`` to disable the ``network`` test resource.

Should we skip tests which "use the network"? But here, it's not a connection to the Internet, but just a test on the loopback interface ("localhost"). Is it also blocked?

A simple test like:

```
HOST = "localhost"

    def test_connect(self):
        port = socket_helper.find_unused_port()
        cli = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        self.addCleanup(cli.close)
        with self.assertRaises(OSError) as cm:
            cli.connect((HOST, port))
        self.assertEqual(cm.exception.errno, errno.ECONNREFUSED)
```

Fails with ETIMEDOUT (error 60) instead of ECONNREFUSED (61):

```
FAIL: test_connect (test.test_socket.NetworkConnectionNoServer.test_connect)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/cirrus-ci-build/Lib/test/test_socket.py", line 5297, in test_connect
    self.assertEqual(cm.exception.errno, errno.ECONNREFUSED)
AssertionError: 60 != 61
```

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -1,0 +1,23 @@
+freebsd_task:
+  freebsd_instance:
+    matrix:
+    - image: freebsd-13-2-release-amd64
+  # Turn off TCP and UDP blackhole.  It is not enabled by default in FreeBSD,
+  # but it is in the FreeBSD GCE images as used by Cirrus-CI.  It causes even
+  # local local connections to fail with ETIMEDOUT instead of ECONNREFUSED.
+  # For more information see https://reviews.freebsd.org/D41751 and
+  # https://github.com/cirruslabs/cirrus-ci-docs/issues/483.
+  sysctl_script:
+    - sysctl net.inet.tcp.blackhole=0
+    - sysctl net.inet.udp.blackhole=0
+  build_script:
+    - mkdir build
+    - cd build
+    - ../configure
+    - make -j$(sysctl -n hw.ncpu)
+  pythoninfo_script:
+    - cd build && make pythoninfo
+  test_script:
+    - cd build
+    # dtrace fails to build on FreeBSD - see gh-73263
+    - make buildbottest TESTOPTS="-j0 -x test_dtrace"

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -20,4 +20,4 @@ freebsd_task:
   test_script:
     - cd build
     # dtrace fails to build on FreeBSD - see gh-73263
-    - make buildbottest TESTOPTS="-j0 -x test_dtrace"
+    - make buildbottest TESTOPTS="-j0 -x test_dtrace --timeout=600"

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -13,7 +13,7 @@ freebsd_task:
   build_script:
     - mkdir build
     - cd build
-    - ../configure
+    - ../configure --with-pydebug
     - make -j$(sysctl -n hw.ncpu)
   pythoninfo_script:
     - cd build && make pythoninfo

--- a/Misc/NEWS.d/next/Tests/2023-09-05-21-42-54.gh-issue-91960.abClTs.rst
+++ b/Misc/NEWS.d/next/Tests/2023-09-05-21-42-54.gh-issue-91960.abClTs.rst
@@ -1,0 +1,1 @@
+FreeBSD 13.2 CI coverage for pull requests is now provided by Cirrus-CI (a hosted CI service that supports Linux, macOS, Windows, and FreeBSD).

